cdap-common/src/main/java/co/cask/cdap/common/namespace/RemoteNamespaceQueryClient.java

```diff
@@ -70,6 +89,10 @@ private InetSocketAddress getNamespaceServiceAddress() {
     if (discoverable != null) {
       return discoverable.getSocketAddress();
     }
-    throw new ServiceUnavailableException(Constants.Service.REMOTE_SYSTEM_OPERATION);
+    throw new ServiceUnavailableException(Constants.Service.APP_FABRIC_HTTP);
+  }
+
+  private HttpRequest addUserIdHeader(HttpRequest request) throws IOException {
+    return new HttpRequest.Builder(request).addHeader(Constants.Security.Headers.USER_ID, userId).build();
```

Comment:
can the ``userId`` be null? What happens when it is null? I think it should be ok, but good to test, otherwise add a guard.